appsec/src/helper/sampler.hpp

```diff
@@ -64,10 +59,20 @@ class sampler {
 
     std::optional<scope> get()
     {
+        if (service_config_->get_asm_enabled_status() ==
+            enable_asm_status::DISABLED) {
+            return std::nullopt;
+        }
+
         const std::lock_guard<std::mutex> lock_guard(mtx_);
 
         std::optional<scope> result = std::nullopt;
 
+        if (sample_rate_ !=
+            valid_sample_rate(service_config_->get_request_sample_rate())) {
+            set_sampler_rate(service_config_->get_request_sample_rate());
```

Comment:
Setting the rate disregards the current number of requests, it should be possible to avoid discarding the current state entirely.

Reply:
I actually did this on purpose. When a new rate is set, then its stats have to change or the rates wont match was neither was set or it is set